client/web/src/LegacyLayout.tsx

```diff
@@ -195,6 +198,15 @@ export const LegacyLayout: FC<LegacyLayoutProps> = props => {
         return <ApplicationRoutes routes={props.routes} />
     }
 
+    // Redirect to post-sign-up page if email is not verified and lead form is not submitted
+    if (
+        props.isSourcegraphDotCom &&
```

Comment:
I'm pretty sure it needs this to fix the recursion.

```suggestion
    if (
        !isPostSignUpPage &&
        props.isSourcegraphDotCom &&
```